zebra/zebra_rib.c

```diff
@@ -2123,6 +2125,19 @@ static void rib_process_result(struct zebra_dplane_ctx *ctx)
 				UNSET_FLAG(old_re->status, ROUTE_ENTRY_FAILED);
 				UNSET_FLAG(old_re->status,
 					   ROUTE_ENTRY_INSTALLED);
+
+				/* Decrement the installed re count for the nhe which is no longer installed
+				 * If there are no more re installed using the old nhe, uninstalled the nhe from dplane */
+				old_re->nhe->installed_re_count--;
```

Comment:
I guess my main concern here would be that we somehow end up with an incorrect count ... should we be testing the count for sanity, or ... ??